docs/start/framework/routing.md

```diff
@@ -301,6 +301,12 @@ You can destructure the `*`, you just have to assign it a new name. A common nam
 const { "*": splat } = params;
 ```
 
+## 404 Catch-All Route
+
+To create a route to catch non-matching routes, you can use the `route("*", "file-path")` route helper.
```

Comment:
````suggestion
You can also use a splat to catch requests that don't match any route:

```ts filename=app/routes.ts lines=[6]
import { type RouteConfig, route,  index } from "@react-router/dev/routes";

export default [
  index("./home.tsx"),
  route("about", "./about.tsx"),
  route("*", "./404.tsx")
] satisfies RouteConfig;
```

````

Let's keep this with the splat section because it already mentions a "catchall" route above. I also added a more fleshed out example of how to use it.